src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -433,9 +433,28 @@ takeWhile1 predicate (Fold fstep finitial fextract) =
 --
 -- /Internal/
 --
-sliceSepBy :: -- MonadCatch m =>
+sliceSepBy :: MonadCatch m =>
     (a -> Bool) -> Parser m a b -> Parser m a b
-sliceSepBy _cond = undefined
+sliceSepBy cond (Parser pstep pinitial pextract) =
+
+    Parser step initial pextract
+
+    where
+
+    initial = pinitial
+
+    step s a =
+        if cond a
+        then do
+            res <- pextract s
+            return $ Done 1 res
+        else do
+            res <- pstep s a
+            case res of
+                Error _ -> do
```

Comment:
I don't think you should extract when an Error is encountered, you should likely just return the Error.